Description: Use the gold linker with memory saving options to avoid build failures caused by lack of RAM
Author: Emmanuel Bourg <ebourg@apache.org>
Forwarded: no
Bug-Debian: https://bugs.debian.org/857464
Taken from: https://sources.debian.org/patches/openjfx/11.0.2+1-1/23-webcore-linker-flags.patch
--- old/modules/javafx.web/src/main/native/Source/WebCore/PlatformJava.cmake
+++ new/modules/javafx.web/src/main/native/Source/WebCore/PlatformJava.cmake
@@ -314,7 +314,7 @@
     set_target_properties(JFXWebKit PROPERTIES LINK_FLAGS "-exported_symbols_list ${WEBCORE_DIR}/mapfile-macosx")
     set(JFXWebKit_EXTERNAL_DEP "${WEBCORE_DIR}/mapfile-macosx")
 elseif (UNIX)
-    set_target_properties(JFXWebKit PROPERTIES LINK_FLAGS "-Xlinker -version-script=${WEBCORE_DIR}/mapfile-vers -Wl,--no-undefined")
+    set_target_properties(JFXWebKit PROPERTIES LINK_FLAGS "-Xlinker -version-script=${WEBCORE_DIR}/mapfile-vers -fuse-ld=gold -Wl,--no-undefined -Wl,--no-keep-memory -Wl,--no-map-whole-files -Wl,--no-keep-files-mapped")
     set(JFXWebKit_EXTERNAL_DEP "${WEBCORE_DIR}/mapfile-vers")
 elseif (WIN32)
     # Adds version information to jfxwebkit.dll created by Gradle build, see JDK-8166265
